### ISSUE TITLE
FYI - contributions - Composite Foreign Keys, Composite indexes

### ISSUE THREAD

**Issue by Trellian** [memtrip/SQLKing#18]
Hi Sam, 

Just to let you know, I'm busy adding support for composite Foreign Keys and indexes, will issue a pull request shortly :)

Cheers,
Adrian

**Comment by samkirton** [comment]
Hi Adrian,

That is great to hear! Could you rebase your changes with the latest master? Last week I added changes for Primary Keys.

Cheers,
Sam

**Comment by Trellian** [comment]
Sam,

Yes, I have already pulled in the Primary Key changes. You pipped me to the post there, I 
had done exactly the same as you for the primary keys.

The way I have structured it, @column no longer has the index= option, they are table 
based now. Are you ok with that?

Regards,
Adrian

**Comment by samkirton** [comment]
hmmm I think I preferred index in @Column, I assume if you have it in @Table you must reference the column by a name string? This has more room for a typo. What do you think?

Sam

**Comment by Trellian** [comment]
I agree, but how else to build composite indexes?

Adrian

**Comment by Trellian** [comment]
I guess it's about the same amount of room for typos as defining indexes in SQL DDL?

Adrian

**Comment by samkirton** [comment]
Would it be possible to have the functionality in both? Basic users can stick to @Column and more advanced users can define their composite as part of @Table?

**Comment by Trellian** [comment]
Yes. I will make it so ;)

Adrian

**Comment by samkirton** [comment]
ok great, also it would be great if you could write some integration tests in the client project that satisfy the implementation :)

**Comment by Trellian** [comment]
I'll give it a shot... I'm new to the integration test scene, and also new to contributing 
via Git, so there might be a few screw-ups along the way that I might need help with. Some 
git concepts still confuse the heck out of me.

I've obviously been following your code for the testing, will atempt to do it the same 
way. I'm a little short on time though, since this is a production project, but I'm really 
enjoying contributing.

I wrote many, many DB templates in Clarion's template language, and I have a bushel full 
of ideas and additions for SQLking.

/Adrian

**Comment by samkirton** [comment]
If you are going to use the new version in production its especially important that we get the integration test coverage, this is something I can write if you are short on time. Have you been running the existing test suite while you have been developing? This will catch any regression issues early on.

Cheers,
Sam

**Comment by Trellian** [comment]
Yes, I agree. I've been running the existing tests, and have also corrected a bug in 
CREATE TABLE that was leaving off the last character before the closing ');' in some cases 
(possibly caused by me)

/Adrian

**Comment by samkirton** [comment]
ok great!
